data_collection/gallica.py

```diff
@@ -0,0 +1,108 @@
+import argparse
+import sys
+from tabulate import tabulate
+import collecting_articles as colart
+import article_extraction as extract
+from tqdm import tqdm
+from multiprocessing import Pool
+import logging
+
+class Gallica :
+    """A CLI class wrapping commands for searching and retrieving articles on gallica
+    """
+
+    def __init__(self):
+        parser = argparse.ArgumentParser(
+            description="A gallica API for retrieving archives",
+            usage=""" gallica command [<args>]
+
+The supported commands for this API are :
+    search      Finds the reference of the given newspaper
+    retrieve    Retrieve all the matching articles
+        """)
+        parser.add_argument("command", help="Subcommand to run")
+        args = parser.parse_args(sys.argv[1:2])
+        if not hasattr(self, args.command):
+            print('Unrecognized command')
+            parser.print_help()
+            exit(1)
+        getattr(self,args.command)(sys.argv[2:])
+
+    def search(self,stringArgs): 
+        parser = argparse.ArgumentParser(
+            description="Finds the reference of the given newspaper"
+        )
+        parser.add_argument("title", help="title of the article to be searched for")
+        parser.add_argument("-n","--number",help="number of results to be displayed",type=int)
+        args = parser.parse_args(stringArgs)
+        n = 10
+        if args.number is not None :
+            n = args.number
+        newspapers = colart.get_newspaper_reference(args.title,n)
+        print(tabulate(newspapers,headers="keys"))
+
+    def retrieve(self,stringArgs):
```

Comment:
needs a space after the ','     
In the different scripts there are some pep8 non compliancies that you can solve with flek-8 - black libraries !